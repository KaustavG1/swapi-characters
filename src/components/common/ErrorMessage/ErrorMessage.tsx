import "./ErrorMessage.css";

function ErrorMessage() {
  return (
    <div className="error-message">An error has occured. Please try again later.</div>
  );
}

export default ErrorMessage;
