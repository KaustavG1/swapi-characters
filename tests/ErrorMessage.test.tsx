import { it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import ErrorMessage from "../src/components/common/ErrorMessage/ErrorMessage";

describe("ErrorMessage component", () => {
  it("renders 'ErrorMessage' component", () => {
    const { container } = render(<ErrorMessage />);
    const errorMessage = screen.getByText(
      /An error has occured. Please try again later./i
    );
    expect(errorMessage).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
