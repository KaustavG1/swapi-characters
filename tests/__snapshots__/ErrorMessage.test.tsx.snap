// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyRecords component > renders 'EmptyRecords' component 1`] = `
<div>
  <div
    class="error-message"
  >
    An error has occured. Please try again later.
  </div>
</div>
`;

exports[`ErrorMessage component > renders 'ErrorMessage' component 1`] = `
<div>
  <div
    class="error-message"
  >
    An error has occured. Please try again later.
  </div>
</div>
`;
